ravenframework/SupervisedLearning/DMD/DMDBase.py

```diff
@@ -0,0 +1,373 @@
+                                                    \item \textit{svd}, randomized single value decomposition
+                                                    \item \textit{correlation\_matrix}, correlation-based reduction.
+                                                  \end{itemize}""", default="svd"))
+    specs.addSub(InputData.parameterInputFactory("reductionRank", contentType=InputTypes.IntegerType,
+                                                 descr=r"""defines the truncation rank to be used for the reduction method.
+                                                 Available options are:
+                                                 \begin{itemize}
+                                                 \item \textit{-1}, no truncation is performed
+                                                 \item \textit{0}, optimal rank is internally computed
+                                                 \item \textit{$>1$}, this rank is going to be used for the truncation
+                                                 \end{itemize}""", default=0))
+    return specs
+
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the model parameter input.
+      @ In, paramInput, InputData.ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+    settings, notFound = paramInput.findNodesAndExtractValues(['pivotParameter','light', 'reductionRank', 'reductionMethod'])
+    # notFound must be empty
+    assert(not notFound)
+    self.settings = {}
+    self.pivotParameterID  = settings.get("pivotParameter")  # pivot parameter
+    self.settings['light'] = settings.get('light')
+    self.settings['reductionMethod'] = settings.get('reductionMethod')
+    self.settings['reductionRank'] = settings.get('reductionRank')
+    if self.pivotParameterID not in self.target:
+      self.raiseAnError(IOError,f"The pivotParameter {self.pivotParameterID} must be part of the Target space!")
+    if len(self.target) < 2:
+      self.raiseAnError(IOError,f"At least one Target in addition to the pivotParameter {self.pivotParameterID} must be part of the Target space!")
+
+
+  def initializeModel(self, dmdParams):
+    """
+      Method to initialize the surrogate model with a dmdParams dictionary
+      @ In, dmdParams, dict, the dictionary containin the parameters/settings to instanciate the model
+      @ Out, None
+    """
+    from pydmd import ParametricDMD
+    from ezyrb import POD, RBF
+
+    assert(self._dmdBase is not None)
+    if self.dmdParams is None:
+      self.dmdParams = dmdParams
+    # intialize dimensionality reduction
+    self._DRrom = POD(self.settings['reductionMethod'], rank=self.settings['reductionRank'])
+    # initialize coefficient interpolator
+    self._interpolator = RBF(kernel='thin_plate_spline', smooth=0, neighbors=None, epsilon=None, degree=None)
```

Comment:
Do you think if there is a benefit here if we use `self` variables to define the default values here?

Reply:
yes you are right. Implemented defaults and settings from input.